components/brave_sync/brave_sync_p3a.cc

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_sync/brave_sync_p3a.h"
+
+#include "brave/components/p3a_utils/bucket.h"
+#include "components/sync/base/user_selectable_type.h"
+
+namespace brave_sync {
+namespace p3a {
+
+void RecordEnabledTypes(bool sync_everything_enabled,
+                        const syncer::UserSelectableTypeSet& selected_types) {
+  using syncer::UserSelectableType;
+  using syncer::UserSelectableTypeSet;
+
+  // "Brave.Sync.EnabledTypes",
+  // 0 - no types or only Bookmarks
+  // 1 - Bookmarks + History
+  // 2 - more than (Bookmarks + History) but less than (Sync All)
+  // 3 - Sync everything
```

Comment:
Have you considered making this an `enum class`. There's [UmaHistogramEnumeration](https://source.chromium.org/chromium/chromium/src/+/main:base/metrics/histogram_functions.h;bpv=1;bpt=1;l=71?q=UmaHistogramExactLinear&ss=chromium%2Fchromium%2Fsrc&gsn=UmaHistogramEnumeration&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fchromium.googlesource.com%2Fcodesearch%2Fchromium%2Fsrc%2F%2Fmain%3Flang%3Dc%252B%252B%3Fpath%3Dbase%2Fmetrics%2Fhistogram_functions.h%23j7iyoHKtSe1QPv-Yb178Iy9gAKfQuXFdhQp-vqUmg-w).

If you think that might look better, make sure your enum has a `kMaxValue`, i.e. `enum class Foo {kFirst, kSecond, kThird, kMaxValue = kThird}`, that way you annotate what is the max value when calling `UmaHistogramEnumeration`

Reply:
Thanks, this notice helped make PR more clear: https://github.com/brave/brave-core/pull/17887/commits/06674f5a4260d7492a25afe2454e6f135f528ed9

Reply:
Ack.